controller/array_action/array_mediator_ds8k.py

```diff
@@ -522,20 +505,25 @@ def _get_api_snapshot(self, snapshot_name, pool_id=None):
                                                                    self.service_address)
         return api_snapshot
 
-    def get_snapshot(self, snapshot_name, pool_id=None):
-        api_snapshot = self._get_api_snapshot(snapshot_name, pool_id)
+    @convert_scsi_id_to_array_id
+    def get_snapshot(self, volume_id, snapshot_name, pool=None):
+        if not pool:
+            source_api_volume = self._get_api_volume_by_id(volume_id)
```

Comment:
please ensure ds8k `get_snapshot` now has a unit test for `IllegalObjectID`

Reply:
+ add it in the interface docstring

Reply:
done